tests/std/tests/P2164R9_views_enumerate/test.cpp

```diff
@@ -0,0 +1,645 @@
+
+        if constexpr (forward_range<R>) {
+            [[maybe_unused]] const iterator_t<R> defaulted;
+        }
+
+        same_as<iterator_t<R>> auto i = r.begin();
+        assert(i.index() == 0);
+
+        { // Check comparisons
+            assert(i == i);
+            assert(!(i < i));
+            assert(i <= i);
+            assert(!(i > i));
+            assert(i >= i);
+            assert(i <=> i == strong_ordering::equal);
+
+            if constexpr (forward_range<R>) {
+                auto i2 = ranges::next(i, 1);
+                assert(i != i2);
+                assert(i < i2);
+                assert(i <= i2);
+                assert(i2 > i);
+                assert(i2 >= i);
+                assert(i <=> i2 == strong_ordering::less);
+                assert(i2 <=> i == strong_ordering::greater);
+            }
+        }
+
+        if constexpr (forward_range<R>) {
+            assert(*i++ == expected[0]);
+        } else {
+            STATIC_ASSERT(is_void_v<decltype(i++)>);
+        }
+        assert(*++i == expected[2]);
+        assert(i.index() == 2);
+
+        if constexpr (bidirectional_range<R>) {
+            assert(*i-- == expected[2]);
+            assert(*--i == expected[0]);
+        }
+
+        if constexpr (random_access_range<R>) {
+            i += 2;
+            assert(*i == expected[2]);
+
+            i -= 2;
+            assert(*i == expected[0]);
+
+            assert(i[2] == expected[2]);
+
+            const auto i2 = i + 2;
+            assert(*i2 == expected[2]);
+
+            const auto i3 = 2 + i;
+            assert(*i3 == expected[2]);
+
+            const auto i4 = i3 - 2;
+            assert(*i4 == expected[0]);
+
+            const same_as<ranges::range_difference_t<V>> auto diff1 = i2 - i;
+            assert(diff1 == 2);
+
+            const same_as<ranges::range_difference_t<V>> auto diff2 = i - i2;
+            assert(diff2 == -2);
+        }
+
+        if constexpr (sized_sentinel_for<sentinel_t<V>, iterator_t<V>>) {
+            const auto i2   = r.begin();
+            const auto sen  = r.end();
+            const auto size = ranges::ssize(expected);
+
+            const same_as<ranges::range_difference_t<V>> auto diff3 = i2 - sen;
+            assert(diff3 == -size);
+
+            const same_as<ranges::range_difference_t<V>> auto diff4 = sen - i2;
+            assert(diff4 == size);
+        }
+
+        using IterMoveResult = tuple<range_difference_t<const Rng>, range_rvalue_reference_t<const Rng>>;
+        [[maybe_unused]] same_as<IterMoveResult> decltype(auto) moved = ranges::iter_move(i);
+        static_assert(noexcept(ranges::iter_move(i))
+                      == (noexcept(ranges::iter_move(i.base()))
+                          && is_nothrow_move_constructible_v<range_rvalue_reference_t<const Rng>>) );
+
+        [[maybe_unused]] same_as<const iterator_t<V>&> decltype(auto) i_base = as_const(i).base();
+        STATIC_ASSERT(noexcept(i.base()));

```

Comment:
It doesn't make a difference for overload resolution, but it seems that the static assert should say `as_const(i).base()` for consistency.